pages/legacy/module_edit.py

```diff
@@ -0,0 +1,103 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+
+import xml.etree.ElementTree as ET
+
+from pages.legacy.base import PrivatePage
+
+from selenium.webdriver.common.by import By
+
+
+class ModuleEdit(PrivatePage):
+
+    URL_TEMPLATE = '/Members/{username}/{module_id}'
+
+    _blank_module_content_string = (
+        '<ns0:content xmlns:ns0="http://cnx.rice.edu/cnxml">\n  '
+        '<ns0:para id="delete_me">\n     \n  </ns0:para>\n</ns0:content>\n\n')
+
+    _title_header_locator = (By.CSS_SELECTOR, '#content div div h1')
+
+    _publish_link_locator = (By.CSS_SELECTOR, 'a[href$="module_publish"]')
+
+    _import_form_locator = (By.CSS_SELECTOR, 'form[action="module_import_form"]')
+    _import_select_locator = (By.CSS_SELECTOR, 'select[name="format"]')
+
+    _content_textarea_locator = (By.ID, 'textarea')
+
+    @property
+    def username(self):
+        return re.search('/Members/([^/]+)/([^/]+)', self.driver.current_url).group(1)
```

Comment:
You may want to add this regex as a property and use `re.compile` and then pass it into the `re.search()` function.
```python
_username_regex = re.compile('/Members/([^/]+)/([^/]+)')
@property
    def username(self):
        return re.search(self_username_regex, self.driver.current_url).group(1)
```